contract/contracts/BoardOfDirectors.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+
+contract BoardOfDirectors is ReentrancyGuardUpgradeable {
+  address[] public owners;
+  address[] public boardOfDirectors;
+  uint256 public actionCount;
+
+  mapping(uint256 => Action) public actions;
+  mapping(uint256 => mapping(address => bool)) public actionApprovals;
+  mapping(uint256 => address[]) public actionApprovers;
+
+  struct Action {
+    address target;
+    string description;
+    uint256 approvalCount;
+    bool isExecuted;
+    bytes data;
+  }
+
+  event BoardOfDirectorsChanged(address[] boardOfDirectors);
+  event OwnersChanged(address[] owners);
+  event ActionAdded(uint256 indexed id, address indexed target, string _description, bytes data);
+  event ActionExecuted(uint256 indexed id, address indexed target, string _description, bytes data);
+  event Approval(uint256 indexed id, address indexed approver);
+
+  function initialize(address[] memory _owners) public initializer {
+    uint256 length = _owners.length;
+    for (uint256 i = 0; i < length; i++) {
+      require(_owners[i] != address(0), 'Invalid owner address');
+    }
+    __ReentrancyGuard_init();
+
+    owners = _owners;
```

Comment:
You can simply call the function setOwners here so you don't need to create a logic for it there.

Reply:
oh we can't because the setOwners have a onlySelf modifier